boxsdk/object/file.py

```diff
@@ -327,13 +328,10 @@ def add_comment(self, message):
             `unicode`
         """
         url = self._session.get_url('comments')
-        message_type = 'tagged_message' if '@[' in message else 'message'
-        data = {
-            message_type: message,
-            'item': {
-                'type': 'file',
-                'id': self.object_id
-            }
+        data = Comment.construct_params_from_message(message)
```

Comment:
I think it might be better to do

```python
CommentClass = self._session.translator.translate('comment')
CommentClass.construct_params_from_message(message)
```

But I don't have strong opinions. I'd be fine if you left it like this for now. It's not that likely that someone will need to override the behavior.